### ISSUE TITLE
Mint XFCE 21 -> 21.2 upgrade successful but shutdown button disabled

### ISSUE THREAD

**Issue by florisvdh** [linuxmint/mintupdate#850]
**Describe the bug**

I could successfully upgrade a Linux Mint XFCE 21 system to version 21.2 with the upgrade tool in mintupdate, but after that the shutdown button was greyed out and did not work.

This appeared to happen because the file `99-mintupdate-temporary.pkla` was still present:

```bash
$ sudo cat /etc/polkit-1/localauthority/90-mandatory.d/99-mintupdate-temporary.pkla
# Blocks shutdown and reboot commands while mintupdate
# automation is running. Gets removed when automation
# finishes.
#
[mintupdate]
Identity=unix-user:*
Action=org.freedesktop.consolekit.system.stop;org.freedesktop.consolekit.system.stop-multiple-users;org.freedesktop.consolekit.system.restart;org.freedesktop.consolekit.system.restart-multiple-users;org.freedesktop.login1.halt;org.freedesktop.login1.halt-ignore-inhibit;org.freedesktop.login1.halt-multiple-sessions;org.freedesktop.login1.reboot;org.freedesktop.login1.reboot-ignore-inhibit;org.freedesktop.login1.reboot-multiple-sessions;org.freedesktop.login1.power-off;org.freedesktop.login1.power-off-ignore-inhibit;org.freedesktop.login1.power-off-multiple-sessions
ResultInactive=no
ResultActive=no

```

So I could solve this locally by deleting this file.

**To Reproduce**

1. Follow the instructions at https://blog.linuxmint.com/?p=4550 to upgrade Mint XFCE 21 to 21.2
2. After announcement of successful upgrade, close the window and try to shutdown the system.
3. The shutdown button appears to be greyed out and does not work.

**Expected behavior**

The shutdown button should not be disabled after the upgrade process.

**Distribution:**

- [X] Linux Mint
- [ ] LMDE
- [ ] Other (please specify)

**Software version:**

6.0.3

**Comment by florisvdh** [comment]
I just encountered the same issue _again_ on the same machine, with `99-mintupdate-temporary.pkla` being present again in 21.2 even while I had removed that file previous time (see above).

Turns out that the reason is an _interrupted automatic software update_ (system configured to update automatically), after which `dpkg --configure -a` needs to be run (recommended by a dialog when trying to update manually). While `dpkg --configure -a` does run successfully and then re-enables further updating, the `99-mintupdate-temporary.pkla` introduced by the interruption stays present, and consequently the shutdown button is disabled.

Maybe you could consider having an extra check in place that will remove `99-mintupdate-temporary.pkla` if appropriate (e.g. on condition that dpkg is not locked).

The observation that the shutdown button was disabled immediately after the 21 -> 21.2 upgrade (above) remains valid, but since the system was and is configured to automatically update, that may be the true reason for this problem.

**Comment by carlosmintfan** [comment]
99-mintupdate-temporary.pkla is not present because of the error but because of the auto-update to avoid the user to shut his system down while it updates. Was it interrupted because you removed that file and shut it down? Maybe there's also another reason for this. Don't know if it's related to the upgrade.

**Comment by florisvdh** [comment]
> Was it interrupted because you removed that file and shut it down?

No, I didn't interrupt auto-updating by removing the file and shutting down. The system was first found in an interrupted state. `dpkg --configure -a` is recommended by mintupdate when it detects an unfinished update, but this doesn't solve (or advise on) the `99-mintupdate-temporary.pkla` still remaining present after running that command. So I did the file removal only to get this finally solved. Hence auto-updating must have been interrupted in some other way before, in a previous session, I cannot think how (I'm not the user of that machine) but anyway the user and/or the software has got it in that state. Since I saw this a second time long after the upgrade to 21.2, it means it can pop up independently of the upgrade it seems.
